src/fsharp/SyntaxTreeOps.fs

```diff
@@ -195,19 +195,30 @@ let rec SimplePatsOfPat synArgNameGenerator p =
     | SynPat.Tuple (false, ps, m)
 
     | SynPat.Paren(SynPat.Tuple (false, ps, _), m) ->
+        let filteredPs =
+            List.choose
```

Comment:
@auduchinok not entirely sure about this part.

Reply:
@nojaf Sorry, I'm late to the party and can't see it anymore. Is it still actual?

Reply:
No, it was a bad idea. The current fix is slightly better.